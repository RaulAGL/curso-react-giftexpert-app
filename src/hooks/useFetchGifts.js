
import  {useState, useEffect} from 'react';
import {getGifts} from "../helpers/getGifts";

export const useFetchGifts = (category) => {

    const [state,setState] = useState({
        data : [], loading : true
    });


    useEffect(() => {

    setTimeout( () => {
        getGifts(category).then(images => {
            setState({
                data: images,
                loading: false
            })
        })
    },5000)
    },[category])

    return state
}
