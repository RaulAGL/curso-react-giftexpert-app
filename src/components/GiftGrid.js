import React from 'react';
import GiftGridItem from "./GiftGridItem";
import {useFetchGifts} from "../hooks/useFetchGifts";
import {PropTypes} from 'prop-types';

const GiftGrid = ({ category}) => {

    const {data, loading} = useFetchGifts(category);

    return (
        <>
            <h3 className="animate__animated animate__backInLeft">{category}</h3>
            { loading && <p className="animate__animated animate__flash">Cargando</p> }
            <div className="card-grid">
                {
                    data.map( (img) => (<GiftGridItem key={img.id} {...img} />    ) )
                }
            </div>
        </>
    )

}

GiftGrid.propTypes = {
    category : PropTypes.string.isRequired
}

export default GiftGrid;


