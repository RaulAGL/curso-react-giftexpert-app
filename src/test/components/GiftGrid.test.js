import GiftGrid from "../../components/GiftGrid";
import {shallow} from 'enzyme';
import React from 'react';
import {useFetchGifts} from '../../hooks/useFetchGifts';

jest.mock('../../hooks/useFetchGifts');

describe('Probar componente <GiftGrid />', () => {

    const category = 'Video Games';
    test('Probar el snapshot del componente', () =>  {

        useFetchGifts.mockReturnValue( { data: [], loading: false } );
        const wrapper = shallow( <GiftGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando cargue el useFetch', () => {
        const gifts = [
            { url : 'https//:domain/test.jpg', title: 'MyTitle', id: 'MyImage' },
            { url : 'https//:domain/test.jpg', title: 'MyTitle', id: 'MyImage2' }
        ];
        useFetchGifts.mockReturnValue( { data: gifts , loading: false } );
        const wrapper = shallow( <GiftGrid category={ category } /> );

        expect( wrapper ).toMatchSnapshot();

        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GiftGridItem').length ).toBe( gifts.length );
    });
});
