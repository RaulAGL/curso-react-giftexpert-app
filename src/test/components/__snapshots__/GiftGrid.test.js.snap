// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar componente <GiftGrid /> Probar el snapshot del componente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInLeft"
  >
    Video Games
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Probar componente <GiftGrid /> debe de mostrar items cuando cargue el useFetch 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInLeft"
  >
    Video Games
  </h3>
  <div
    className="card-grid"
  >
    <GiftGridItem
      id="MyImage"
      key="MyImage"
      title="MyTitle"
      url="https//:domain/test.jpg"
    />
    <GiftGridItem
      id="MyImage2"
      key="MyImage2"
      title="MyTitle"
      url="https//:domain/test.jpg"
    />
  </div>
</Fragment>
`;
